File: python/day02.py
def versus(left, right):
    if left == "A" and right == "X":
        return 3
    if left == "A" and right == "Y":
        return 6 
    if left == "A" and right == "Z":
        return 0
    if left == "B" and right == "X":
        return 0
    if left == "B" and right == "Y":
        return 3 
    if left == "B" and right == "Z":
        return 6 
    if left == "C" and right == "X":
        return 6
    if left == "C" and right == "Y":
        return 0
    if left == "C" and right == "Z":
        return 3

def versus2(left, right):
    if left == "A" and right == "X":
        return 3
    if left == "A" and right == "Y":
        return 1 
    if left == "A" and right == "Z":
        return 2
    if left == "B" and right == "X":
        return 1
    if left == "B" and right == "Y":
        return 2 
    if left == "B" and right == "Z":
        return 3 
    if left == "C" and right == "X":
        return 2
    if left == "C" and right == "Y":
        return 3
    if left == "C" and right == "Z":
        return 1

def part1():
    with open('../data/day02.txt') as f:
        data = f.read().splitlines()

    play_scores = { 'X': 1, 'Y': 2, 'Z': 3 }

    total = 0
    scores = []

    for val in data:
        v = val.split(" ")
        score = versus(v[0], v[1])
        scores.append(score + play_scores[v[1]])

    answer = sum(scores)

    print(f'Total round one score is {answer}')


    f.close()

def part2():
    with open('../data/day02.txt') as f:
        data = f.read().splitlines()

    play_scores = { 'X': 0, 'Y': 3, 'Z': 6 }

    total = 0
    scores = []

    for val in data:
        v = val.split(" ")
        score = versus2(v[0], v[1])
        scores.append(score + play_scores[v[1]])

    answer = sum(scores)

    print(f'Total round two score is {answer}')


    f.close()


part1()
part2()

File: javascript/src/day-02.js
import { readFileSync } from "fs";

// COLUMN 1
// A for Rock
// B for Paper
// C for Scissors

// COLUMN 2
// X for Rock
// Y for Paper
// Z for Scissors

// 0 if you lost
// 3 if the round was a draw
// 6 if you won

// 1 rock
// 2 paper
// 3 scissors

function roundOne() {
  let playValue = {
    X: 1,
    Y: 2,
    Z: 3,
  };

  function versus(left, right) {
    let v =
      left === "A" && right === "X"
        ? 3
        : left === "A" && right === "Y"
        ? 6
        : left === "A" && right === "Z"
        ? 0
        : left === "B" && right === "X"
        ? 0
        : left === "B" && right === "Y"
        ? 3
        : left === "B" && right === "Z"
        ? 6
        : left === "C" && right === "X"
        ? 6
        : left === "C" && right === "Y"
        ? 0
        : left === "C" && right === "Z"
        ? 3
        : null;
    return v;
  }

  const data = readFileSync("../data/day02.txt", "utf8");
  const lines = data.split("\n");

  let scores = [];

  let total = 0;

  while (lines.length) {
    const val = lines.shift();

    let [user, me] = val.split(" ");

    let score = versus(user, me);
    total += score + playValue[me];
    scores.push(score + playValue[me]);
  }

  console.log(`My total round one score: ${total}`);
}

function roundTwo() {
  let playValue = {
    X: 0,
    Y: 3,
    Z: 6,
  };

  // X lose
  // Y draw
  // Z win

  // 1 rock
  // 2 paper
  // 3 scissors
  function versus(left, right) {
    let v =
      left === "A" && right === "X"
        ? 3
        : left === "A" && right === "Y"
        ? 1
        : left === "A" && right === "Z"
        ? 2
        : left === "B" && right === "X"
        ? 1
        : left === "B" && right === "Y"
        ? 2
        : left === "B" && right === "Z"
        ? 3
        : left === "C" && right === "X"
        ? 2
        : left === "C" && right === "Y"
        ? 3
        : left === "C" && right === "Z"
        ? 1
        : null;
    return v;
  }

  const data = readFileSync("../data/day02.txt", "utf8");
  const lines = data.split("\n");

  let scores = [];

  let total = 0;

  while (lines.length) {
    const val = lines.shift();

    let [user, me] = val.split(" ");

    let score = versus(user, me);
    total += score + playValue[me];
    scores.push(score + playValue[me]);
  }

  console.log(`My total round two score: ${total}`);
}

roundOne();
roundTwo();
